### PR TITLE
Movielens dataloader

### PR DESCRIPTION
**Pull request by zkid18**
## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)?
- [x] Did you check the code style? `catalyst-make-codestyle && catalyst-check-codestyle` (`pip install -U catalyst-codestyle`).
- [x] Did you make sure to update the docs? We use Google format for all the methods and classes.
- [x] Did you check the docs with `make check-docs`?
- [x] Did you write any new necessary tests?
- [x] Did you add your new functionality to the docs?
- [x] Did you update the [CHANGELOG](https://github.com/catalyst-team/catalyst/CHANGELOG.md)?
- [x] **You can use 'Login as guest' to see Teamcity build logs.**

## Description

The PR motivation to enrich catalyst dataset with movie-lens-100k for recsys purposes. 
The PR is still under development so any suggestions and comments appreciated. 

I tired to follow the MNIST [example](https://github.com/catalyst-team/catalyst/blob/f99e08b739bafa668276cd0da2c5fb2d1e430d6e/catalyst/contrib/datasets/mnist.py) but seems that PyTorch's Dataset is not an optimal solution for the movielens. 

For interaction matrix we need an abstract class with 
```
    def __getitem__(self, item_index, user_index):
        raise NotImplementedError

    def __len__(self):
        """ (user_dim, item_dim) """
        raise NotImplementedError
```
While PyTorch Dataset [provides](https://pytorch.org/docs/stable/_modules/torch/utils/data/dataset.html#Dataset
) support  the single way of sampling data from the dataset given an index.

This PR describes the download and parsing part, but doesn't compile with PyTorch protocol. 

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves an existing feature)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

**Comment by zkid18** [comment]
Some ideas towards data representation
https://medium.com/@rinabuoy13/explicit-recommender-system-matrix-factorization-in-pytorch-f3779bb55d74
> First, I create a dataset class which inherits from torch.utils.data.Dataset. The constructor takes two numpy arrays — 2D user-item id array and 1D rating array. I implement `__len__` method which returns the length of dataset, and `__getitem__` which returns tensors of user-item id pairs and ratings.

```
class RatingDataset(Dataset):
  def __init__(self, train, label):
    self.feature_= train
    self.label_= label
  def __len__(self):
    #return size of dataset
    return len(self.feature_)
  def __getitem__(self, idx):
    return              torch.tensor(self.feature_[idx]),torch.tensor(self.label_[idx])
```

As for another abstract class we can refer this implementation https://github.com/maciejkula/spotlight/blob/master/spotlight/interactions.py
https://github.com/maciejkula/spotlight/blob/master/spotlight/datasets/movielens.py

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Added
 
+- `MovieLens dataset` loader [#903](https://github.com/catalyst-team/catalyst/pull/903)
```
Comment (Scitator) [inline CHANGELOG.md]: looks like we need to change this :)

**Review by Scitator — DISMISSED** [review]
Looks good, but we need to fix codestyle first :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Added
 
+- `MovieLens dataset` loader ([#903](https://github.com/catalyst-team/catalyst/pull/903))
```
Comment (Scitator) [inline CHANGELOG.md:12]: looks like we need to move it up :)

**Pull request merged**